Core/Src/Sensors/AnalogSensor.c

```diff
@@ -1,7 +1,187 @@
 #include "../../Inc/Sensors/AnalogSensor.h"
+#include <stdio.h>
+
+void initAnalogSensor(AnalogSensor* sensor, const char* name, int hz, int channel) {
+    initSensor(&sensor->base, name, hz);
+    sensor->channel = channel;
+
+    if(sensor->channel == 0) {
+    	// HAL code to bind to PA0
+    	__HAL_RCC_GPIOA_CLK_ENABLE();
```

Comment:
There are statements like this that could be applied in ranges to make it make it more readable. ie. 
```c
if (sensor->channel <= 7) {
        __HAL_RCC_GPIOA_CLK_ENABLE();
        HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
...
```